medcat/utils/preprocess_snomed.py

```diff
@@ -107,6 +113,8 @@ def to_concept_df(self):
             contents_path = os.path.join(paths[i], "Snapshot", "Terminology")
             concept_snapshot = "sct2_Concept_Snapshot"
             description_snapshot = "sct2_Description_Snapshot-en"
+            if self.au_ext:
+                description_snapshot += "-AU"
             if self.uk_ext:
```

Comment:
should this be a `elif` as it seems like a `ValueError` if  `uk_ext=True ` and `au_ext=True` right?

Reply:
In principle, you're right.
And the same would be true for `uk_ext` and `uk_drug_ext`, actually. Even though semantically you might not think that. They're distinctly different strategies for getting the snapshot/description/relationship file name parts.

The above also suggests the `ValueError` should actually be raised if more than one of the three (`uk_ext`, `uk_drug_ext`, or `au_ext`) are set.

However, treating these as a "only one must be set" would suggests changes in the `list_all_relationships` and `relationship2json` methods as well. Which I feel would be outside the scope of this PR.

So all in all, I've tried to make a minimal possible PR. At least for now.


Though in principle, an overall better structure would be beneficial here.
Perhaps a single `Enum` init argument for extension (ranging from the default of international, to UK extension, UK Drug extension, and now AU extension).
And then use a method (or even a `dict`) to find the values of `concept_snapshot`, `description_snapshot`, and `relationship_snapshot` once depending on the specific extension.
Though this could have issues where the `uk_ext` attribute is set after initialisation (I think I recall seeing something like that somewhere).

The above shouldn't really be that difficult in principle. But again, would fall outside the scope of this PR.

Reply:
okay agreed and yes makes sense. Bigger change to an enum and more branching of code for support of more countries likely when we get a handle on further country specific releases.